### ISSUE TITLE
Add statistics information for individual checks (especially time of execution)

### ISSUE THREAD

**Issue by potiuk** [pre-commit/pre-commit#1144]
I would love to have an option to add some statistics to be displayed for each check. This could be very useful for running pre-commit in CI. I had a case recently of a check that run accidentally a bit longer than expected (it was building unnecessary docker image) and I have not realised that it is taking longer - because we had other checks after and I could only see the total execution time.

Also some other useful information that we might see is how many parallel processes were run and how many files were passed as parameters. This might be really useful especially for people who do not understand that pre-commit runs in parallel by default  - this can have some undesireable side effects if you forget to set "require_serial" to True when you need. And seeing that in the output of CI might immediately show that something is wrong. For now the output is a bit of "black-box".

An option to see some additional information (`--add-statistics`) might be super-useful.

Example output I imagine (maybe a bit better aligned):

```
Check if image build is needed...........................................Passed [23.5s, 1 process, no-files]  
Add licence for all JINJA template files.................................Passed [2.25s, 8 processes, 48 files]
Check Shell scripts syntax correctness...................................Passed [6.00s, 8 processes, 10 files]
Lint dockerfile..........................................................Passed [2.20s, 3 processes, 3 files]
Run mypy.................................................................Passed [12.00s, 8 processes, 1456 files]
Run pylint for main sources.............................................Skipped 
Run pylint for tests....................................................Skipped
Run flake8...............................................................Passed [24.05s, 8 processes, 1456 files]
```

**Comment by potiuk** [comment]
Happy to contribute it if there is an agreement to add it :)

**Comment by asottile** [comment]
the output is intentionally quiet and I don't want to change that, your proposal also makes things no longer line up

maybe just add this to verbose if at all ?

**Comment by potiuk** [comment]
verbose is .... far too verbose. It's great to see only "passed" but then having some stats, would make it also actionable for cases that are "valid properties" of the check rather than just passed/failed status. Most of the UnitTest frameworks have such feature to display for example the timing. 

And it is perfectly ok to have a separate switch for that and do not have it in the output by default. I think it might be really useful for the CI case I described.

**Comment by asottile** [comment]
I don't want a combinatoric explain of options, especially ones I'm not going to use

**Comment by asottile** [comment]
if it's only for CI who cares how verbose it is?

**Comment by potiuk** [comment]
I think it might be useful for local pre-commits as well, but the main usage scenario for this flag I foresee for CI case. We are currently using it with `--all-files --show-diff-on-failure` and it works great (minimum output + we see exact diffs when diff-failure happens). 

BTW. I have found that having CI build for pre-commit is great promotor of pre-commit in general - because when something fails, installing pre-commit and running that failed check is simplest way to do (and pre-commit nicely suggests that action). But CI is also good to observe how good your build is.

In this case I only found out that we have a problem because the build steps were running for a really long time and failed (but I would immediately see the problem if I had the stats).
In my particular case we have the "build needed check" (see the #1143 ) and when it does not need to rebuild the image it will be "passed" in ms. But if (happened in my case) due to bug in build need  detection it took maybe 2-3 minutes. - I would immediately see this with statistics but in case of our build the only thing i could see is "passed" without knowing how much time this particular check took.

**Comment by asottile** [comment]
I don't want another option, let's add it to verbose output

**Comment by blueyed** [comment]
I think just having "Passed in x.xs" by default (instead of just "Passed") would be useful to have by default, which could easily be aligned (assuming that nothing takes 100 seconds).
(Not sure about the extra stats (processes / files), likely they should go into "verbose" mode - never used that, it is too verbose by default already for me (77aab5e))

**Comment by asottile** [comment]
please read the discussion -- I've already said no to that

**Comment by blueyed** [comment]
@asottile 
As far as I can see you've said "no" to "Passed [23.5s, 1 process, no-files]" only.
Anyway, just my 2c - you've duped it after all.

**Comment by blueyed** [comment]
This is kind of offtopic (more related to #957): in general I find "Running pre-commit: ……✔✔…✔………………… (3 passed, 10 skipped in 1.0s)" enough information, but in non-quiet mode it would be nice to display durations also with subtasks.

**Comment by asottile** [comment]
please stay on topic, thanks

**Comment by blueyed** [comment]
i> please stay on topic, thanks

Please don't delete my comments, thanks.  This really makes me stay away from you / this repo now.

FWIW:have 
You've a) duped a more specific issue here, and then b) deleted a comment from me speciying how this (durations) interacts with display (verbose or not) in general.  :shrug:

You've also not answered my correction about me telling you that your "please read the discussion -- I've already said no to that" comment was not fitting.
That would have been helpful with keeping this on-topic (since you've duped it here in the first place).
:shrug:

**Comment by asottile** [comment]
This has been released as part of [v1.21.0](https://github.com/pre-commit/pre-commit/releases/tag/v1.21.0) -- thanks again for the issue 🎉!
